FEXCore/Source/Utils/CPUInfo.cpp

```diff
@@ -14,17 +15,31 @@
 namespace FEXCore::CPUInfo {
 #ifndef _WIN32
 uint32_t CalculateNumberOfCPUs() {
-  char Tmp[PATH_MAX];
+  static uint32_t NumCPUs {};
+  if (NumCPUs) {
+    return NumCPUs;
+  }
+
+#define CPUSTRING "/sys/devices/system/cpu/cpu"
+  constexpr size_t CPUStringLength = std::char_traits<char>::length(CPUSTRING);
+
+  // Support up to 8 digits of CPU cores plus the null terminator.
+  char Tmp[CPUStringLength + 1 + 8] = CPUSTRING;
+#undef CPUSTRING
```

Comment:
This got overly complex. `std::string_view` makes it more straightforward:

```suggestion
  // Support up to 8 digits of CPU cores plus the null terminator.
  char Tmp[CPUString.size() + 1 + 8];
  constexpr std::string_view CPUString = "/sys/devices/system/cpu/cpu";
  std::ranges::copy(CPUString, Tmp);
```

Reply:
constexpr also has the academic benefit of knowing the string size at compile time instead of strlen getting called at runtime with this suggested implementation.

```
        mov     x0, sp
        ldp     q0, q1, [x8]
        mov     x20, sp
        stp     q0, q1, [sp]
        ldp     q0, q2, [x8, #32]
        stp     q0, q2, [sp, #32]
        bl      strlen
 ```

Reply:
Did you enable compiler optimizations? That would be the relevant scenario to reference when optimizing in the nano-second range.

Reply:
This is with -O3, so yes.

Reply:
Interesting, I tested against gcc and it optimizes away strlen there while clang doesn't. Since it's just one-time setup, this probably has no significant overhead compared to the filesystem accesses below, but happy to just go with the former proposal if you prefer.

Reply:
Not quite as optimal since clang zero pads the ending of the string when it is a char array which lets it generate the whole string in a single pair of 256-bit ldp and stp instructions but converted over to the string_view anyway.